packages/ui/src/components/PropertiesModal/PropertiesModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { Modal } from '@patternfly/react-core';
+import { Caption, Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { FunctionComponent } from 'react';
+import { camelComponentToTable, kameletToTable } from '../../camel-utils/camel-to-table.adapter';
+import { CatalogKind, ICamelComponentDefinition, ICamelProcessorDefinition, IKameletDefinition } from '../../models';
+import { ITile } from '../Catalog';
+import { EmptyTableState } from './EmptyTableState';
+import { IPropertiesTable, PropertiesHeaders } from './PropertiesTable.models';
+
+interface IPropertiesModalProps {
+  tile: ITile;
+  onClose: () => void;
+  isModalOpen: boolean;
+}
+
+export const PropertiesModal: FunctionComponent<IPropertiesModalProps> = (props) => {
+  let table: IPropertiesTable;
+  switch (props.tile.type) {
+    case CatalogKind.Component:
+      table = camelComponentToTable(props.tile.rawObject as ICamelComponentDefinition);
+      break;
+    case CatalogKind.Processor:
+      table = camelComponentToTable(props.tile.rawObject as ICamelProcessorDefinition);
+      break;
+    case CatalogKind.Kamelet:
+      table = kameletToTable(props.tile.rawObject as IKameletDefinition);
+      break;
+    default:
+      throw Error('Unknown CatalogKind during rendering modal: ' + props.tile?.type);
+  }
+
+  return (
+    <Modal title={props.tile.title} isOpen={props.isModalOpen} onClose={props.onClose} ouiaId="BasicModal">
+      <p>{props.tile.description}</p>
+      <Table aria-label="Simple table" variant="compact">
+        <Caption>{'Available properties (' + table.rows.length + ')'}</Caption>
+        <Thead>
+          <Tr>
+            {table.headers.includes(PropertiesHeaders.Name) && <Th>{PropertiesHeaders.Name}</Th>}
```

Comment:
Considering that the `headers` property is an array, wouldn't be easier to iterate over it and make the text title-cased using CSS? :thinking:

Reply:
yes, it can be done by 
```
  <Thead>
    <Tr>
      {table.headers.map((header) => (<th key={header}>{header}</th>))}
    </Tr>
  </Thead>
```
however, we need to be sure that table.headers order in array is the same as in table.rows.

Reply:
Or we can delete table.headers, and get headers dynamically from the first row, e.g.
```
  const header = Object.keys(table.rows[0])
```
but in that case, the table should contain at least one row

Reply:
refactored and force push